### PR TITLE
Add explanation to CONTRIBUTING for where to find other docs

### PR DESCRIPTION
**Pull request by SeanKilleen**
Based on my own difficulty finding the source of the docs I was looking for as a first-time contributor.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,3 +14,5 @@ Prometheus uses GitHub to manage reviews of pull requests.
   on our [mailing list](https://groups.google.com/forum/?fromgroups#!forum/prometheus-developers).
   This will avoid unnecessary work and surely give you and us a good deal
   of inspiration.
+
+* Note that not all of the documents you see on <https://prometheus.io> are sourced from this repository. If you're unable to find the documents here, you may want to look in <https://github.com/prometheus/prometheus/tree/main/docs>.
```
Comment (bwplotka) [inline CONTRIBUTING.md:18]:
```suggestion
* Generally, the documents you see on https://prometheus.io are sourced from this repository. The only exception are the documentation pages that are versioned per Prometheus release. Those are sourced in [prometheus/prometheus docs directory](https://github.com/prometheus/prometheus/tree/main/docs) for the related version.

```

**Review by bwplotka — APPROVED** [review]
Thanks! Small suggestion to be more precise, WDYT?

File: CONTRIBUTING.md (hunk 1 above)
Comment (SeanKilleen) [reply CONTRIBUTING.md:18]: 👍 That offers both a a helpful explanation to me and a good improvement. I'll make one more tweak after I commit it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -14,3 +14,6 @@ Prometheus uses GitHub to manage reviews of pull requests.
   on our [mailing list](https://groups.google.com/forum/?fromgroups#!forum/prometheus-developers).
   This will avoid unnecessary work and surely give you and us a good deal
   of inspiration.
+
+* Generally, the documents you see on https://prometheus.io are sourced from this repository. The only exception are the documentation pages that are versioned per Prometheus release. Those are sourced in [prometheus/prometheus docs directory](https://github.com/prometheus/prometheus/tree/main/docs) for the related version.
```
Comment (SeanKilleen) [inline CONTRIBUTING.md:18]:
```suggestion
* Generally, the documents you see on <https://prometheus.io> are sourced from this repository. The only exception are the documentation pages that are versioned per Prometheus release. Those are sourced in [the docs directory of the prometheus/prometheus repository](https://github.com/prometheus/prometheus/tree/main/docs) for the related version.
```

**Pull request merged**